src/System.Xml.XDocument/System/Xml/Linq/XComment.cs

```diff
@@ -92,6 +94,21 @@ public override void WriteTo(XmlWriter writer)
             writer.WriteComment(value);
         }
 
+        /// <summary>
+        /// Write this <see cref="XComment"/> to the passed in <see cref="XmlWriter"/>.
+        /// </summary>
+        /// <param name="writer">
+        /// The <see cref="XmlWriter"/> to write this <see cref="XComment"/> to.
+        /// </param>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        public override async Task WriteToAsync(XmlWriter writer, CancellationToken cancellationToken)
+        {
+            if (writer == null) throw new ArgumentNullException("writer");
```

Comment:
Yes please ignore my comments on the one-line if statements as you are correct all this code still needs to be run through our formatting tool, we have issue https://github.com/dotnet/corefx/issues/496 tracking that.